client/app/states/catalogs/details/details.state.js

```diff
@@ -13,64 +14,94 @@ function getStates() {
-    vm.dialogs = dialogs.resources[0].content;
-  }
-
-  vm.addToCart = addToCart;
-  vm.cartAllowed = ShoppingCart.allowed;
-  vm.alreadyInCart = alreadyInCart;
-  vm.addToCartDisabled = addToCartDisabled;
-
   var autoRefreshableDialogFields = [];
   var allDialogFields = [];
-
-  DialogFieldRefresh.setupDialogData(vm.dialogs, allDialogFields, autoRefreshableDialogFields);
-
-  var dialogUrl = 'service_catalogs/' + serviceTemplate.service_template_catalog_id + '/service_templates';
-
-  angular.forEach(allDialogFields, function(dialogField) {
-    dialogField.refreshSingleDialogField = function() {
-      DialogFieldRefresh.refreshSingleDialogField(allDialogFields, dialogField, dialogUrl, vm.serviceTemplate.id);
+  vm.init = init;
+  let dialogs = {};
+  const serviceTemplate = {};
+  let serviceRequest = {};
+
+  function init() {
+    vm.loading = true;
+    vm.addToCart = addToCart;
+    vm.cartAllowed = ShoppingCart.allowed;
+    vm.addToCartEnabled = false;
+    vm.alreadyInCart = alreadyInCart;
+    vm.addToCartDisabled = addToCartDisabled;
+
+    vm.dialogData = {};
+
+    const serviceRequestPromise = () => {
+      return new Promise((resolve, _reject) => {
+        if ($stateParams.serviceRequestId) {
+          CollectionsApi.get('requests', $stateParams.serviceRequestId, {}).then((data) => { resolve(data); });
+        } else {
+          resolve(false);
+        }
+      });
     };
-  });
+    serviceRequestPromise().then((resolvedServiceRequest) => {
+      serviceRequest = resolvedServiceRequest;
+
+      const dialogRequest = new Promise((resolve, _reject) => {
+        const options = { expand: 'resources', attributes: 'content' };
+        let serviceTemplateId = $stateParams.serviceTemplateId;
+        if (!serviceTemplateId) {
+          serviceTemplateId = serviceRequest.source_id;
+        }
+        CollectionsApi.query('service_templates/' + serviceTemplateId + '/service_dialogs', options).then((resolvedDialogs) => {
+          resolve(resolvedDialogs);
+        });
+      });
 
-  AutoRefresh.listenForAutoRefresh(
-    allDialogFields,
-    autoRefreshableDialogFields,
-    dialogUrl,
-    vm.serviceTemplate.id,
-    DialogFieldRefresh.refreshSingleDialogField
-  );
+      const serviceTemplateRequest = new Promise((resolve, _reject) => {
+        let serviceTemplateId = $stateParams.serviceTemplateId;
+        if (!serviceTemplateId) {
+          serviceTemplateId = serviceRequest.source_id;
+        }
+        var options = { attributes: ['picture', 'picture.image_href'] };
+        CollectionsApi.get('service_templates', serviceTemplateId, options).then((data) => {
+          resolve(data);
+        });
+      });
+      const allPromises = [dialogRequest, serviceTemplateRequest];
+      Promise.all(allPromises).then((data) => {
+        const SERVICE_TEMPLATE_RESPONSE = 1;
+        const DIALOGS_RESPONSE = 0;
+        dialogs = data[DIALOGS_RESPONSE];
+        vm.dialogs = dialogs.resources[0].content;
```

Comment:
<img width="1239" alt="screen shot 2017-12-14 at 7 55 07 am" src="https://user-images.githubusercontent.com/6640236/33993417-750765dc-e0a4-11e7-9826-8760947fd45b.png">

Seeing this error when selecting a service, might want to test this value before assignment :-/

Then again maybe this is just an instance of a borked service catalog item 🤷‍♀️